semantics/src/main/kotlin/com/strumenta/kolasu/semantics/node/repository/NodeRepository.kt

```diff
@@ -0,0 +1,9 @@
+package com.strumenta.kolasu.semantics.node.repository
+
+import com.strumenta.kolasu.model.Node
+
+interface NodeRepository {
+    fun load(identifier: String): Node?
+
+    fun store(node: Node): String?
```

Comment:
When do we need to store nodes?

Reply:
Good question! :) Now that the symbol resolution does not need to retrieve nodes for global symbols, we never need access to a node repository - it was removed in the last commit.